posts/h5ad_conversion/index.qmd

```diff
@@ -0,0 +1,612 @@
+  Seurat::DarkTheme()
+```
+![](figures/Fig-mBrain-S1S2-cellType-slidesArranged.png)
+
+
+## Clean up to keep only necessary data {#sec-cleanUp}
+
+Next, we would further clean up the `Seurat` object (single-slide object from [Case 1 in section -@sec-perSlide] or full-study-level object from [Case 2 in section -@sec-perStudy]) by trimming it down to contain only the data of interested. 
+
+Typically, one would keep the raw data counts from `RNA` assay (this example dataset uses `Nanostring` as assay name for `RNA`), cell embedding for `umap` (standard AtoMx exported object uses `approximateumap` as name for `umap`). We would also store the spatial coordinates of each cell as the cell embedding in a dimension reduction object called `spatial`.
+```{r dietSeurat, message=FALSE, eval = FALSE}
+# clean up seurat object to only necessary data 
+seu.obj2 <- Seurat::DietSeurat(
+  seu.obj1, 
+  
+  # subset of assays to keep, standard AtoMx object uses "RNA" assay
+  # of note, AtoMx stores normalized RNA counts in separate "RNA_normalized" assay
+  assays = "Nanostring",
+  
+  # keep raw counts stored in "counts" layer
+  # use "data" or "scale.data" if prefer to keep normalized counts before or after scaling
+  layers = "counts", 
+  
+  # dimension reduction to keep, standard AtoMx object uses "approximateumap"
+  dimreducs = "umap")
+
+# clear the `images` slot
+allImgs <- names(seu.obj1@images)
+for (img in allImgs){
+  seu.obj2[[img]] <- NULL
+}
+
+# add in spatial coordinates for current slide or study as a new dimension reduction
+colnames(spatial_coords) <- paste0("SPATIAL_", seq_len(ncol(spatial_coords)))
+seu.obj2[["spatial"]] <- Seurat::CreateDimReducObject(
+  embeddings = spatial_coords, 
+  key = "SPATIAL_", 
+  # standard AtoMx object use "RNA" assay
+  assay = "Nanostring")
+
+```
+
+## Converting to `h5ad` format via `h5Seurat`
+
+Lastly, we would export the cleaned `Seurat` object into `h5Seurat` format and then further convert it into `h5ad` format using `SeuratDisk::Convert` function.
+```{r h5Seurat, message=FALSE, eval = FALSE}
+# export as "h5Seurat" object
+SeuratDisk::SaveH5Seurat(seu.obj2, 
+                         filename = paste0(slideName, "_subset.h5Seurat"))
+```
+<code><span style="color: red;">
+Creating h5Seurat file for version 3.1.5.9900<br/>
+Adding counts for Nanostring<br/>
+Adding data for Nanostring<br/>
+No variable features found for Nanostring<br/>
+No feature-level metadata found for Nanostring<br/>
+Adding cell embeddings for umap<br/>
+No loadings for umap<br/>
+No projected loadings for umap<br/>
+No standard deviations for umap<br/>
+No JackStraw data for umap<br/>
+Adding cell embeddings for spatial<br/>
+No loadings for spatial<br/>
+No projected loadings for spatial<br/>
+No standard deviations for spatial<br/>
+No JackStraw data for spatial</span></code>
+
+```{r h5adConversion, message=FALSE, eval = FALSE}
+# convert to h5ad format
+SeuratDisk::Convert(paste0(slideName, "_subset.h5Seurat"), 
```

Comment:
Do you need to specify a location for where to store this h5ad file?

Reply:
no, it always saved in your working directory. I think it doesn't work if you want to save it elsewhere. Let me put a note in that comment line about this.